### ISSUE TITLE
[Accessibility]: Search box and homepage misc mobile bugs

### ISSUE THREAD

**Issue by limitedmage** [sourcegraph/sourcegraph#34727]
### Audit type

Mobile device navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34188 and https://github.com/sourcegraph/sourcegraph/issues/34189

### Problem description

- [ ] Search context dropdown: Description of search contexts almost completely cut off (suggested fix: add a tooltip for the description)
![image](https://user-images.githubusercontent.com/206864/166075973-b066be09-329b-4f74-a473-bc907282a272.png)

### Expected behavior

Issues should be fixed

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [X] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [X] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by GitStart. I have assigned the label 'gitstart' and added the assignee 'gitstart-sourcegraph'.

**Comment by sourcegraph-bot-2** [comment]
Heads up @benvenker @lguychard - the "team/search-product" label was applied to this issue.

**Comment by limitedmage** [comment]
@gitstart-sourcegraph I've updated the issue with a different suggested fix, it's ready to start implementation.
